auth0/nextjs-auth0
# beta

- Initial commit of Beta branch
- Add frontend hook
- Export hook, UserProvider and UserProfile
- Use default export
- Add browser exports
- Update api-call-example
- Remove Layout props
- Update basic-example
- Update typescript-example
- Address review feedback
- Fix lint
- Add frontend hook tests
- Fix json union type
- Remove now unnecessary rule
- Rename mock type
- Callback tests wip
- Add integration tests [SDK-2059] (#190)
- TransientStore/cookiestore tests wip
- Updates for parallelizing tests
- Telemetry and Client test
- Config test
- Missing coverage
- Once over
- Flakey ts integration test
- Refactor Next.js handlers
- Fix tests
- Fix tests
- revert trailing comma updates
- createHandlers -> handleAuth for consistency with other handlers
- Fix examples
- Change signature of `withPageAuthRequired` since both arguments are optional
- fix tests
- Use Next.js 10.x redirecting behaviour
- Add option to use named exports that create singleton instance
- Update new secure cookie behaviour form auth0/express-openid-connect#159
- Update next.js handler tests
- debug circleci test failures
- remove debugging
- test removing ts example from tests
- Run a real Next.js server
- Logout handler
- Remaining handler tests
- Added architecture documents [SDK-2066] (#204)
- Dry up the handler tests a little
- DRY up the handler tests a little
- Added helpers tests
- Add get-access-token tests
- lint/test scripts
- Attempting to fix cypress tests
- fix some missing coverage
- add missing cov
- fix tests
- Implement HOC
- Add tests and rename to withCSRAuthRequired
- Add doc comments
- Re-add type annotation
- Rename test symbols
- Removed unused dev dependency
- Move api-call-example in typescript-example
- Rename typescript-example to kitchen-sync-example
- Make the basic application minimal
- Only run cypress on kitchen-sync
- Protect Client Side Rendered Page
- Remove protected page
- Protect TV Shows API
- Protect shows page and update navigation
- Fix typo
- Revert incorrect merge
- Revert incorrect merge
- Drop withAuthenticationRequired
- With page auth required (server)
- Add/fix tests
- rename/fix examples
- Export types from index.ts
- Move example API to Next
- Add TS docs
- add coverage gate
- Remove some warnings, add a comment about the external api
- Update generated docs, README, add EXAMPLES
- README updates
- Doc updates
- Update README.md
- Remove unnecessary `else`s
- Apply suggestions from code review
- Update README.md
- Update README.md
- Remove express specifics from docs
- Add SessionCache unit tests
- Add extra SessionCache unit test case
- Prepare package for release, fix the basic example
- Prepare application to be deployable with Vercel
- Do not prefix base url when not defined
- Fix doc commet snippets
- Fix punctuation
- Fix version test, `initAuth0` houldn't throw in browser mode, add test for baseURL
- Add `httpTimeout` option to match v0.x
- Add missing option to configure callback
- Update examples per @jamesqquick suggestions
- Tweak _app wrapper code sample
- Tweaked user authentication sample
- Correct the callback URL in Auth0 setup
- Separate config sample in generated docs
- Split out the code comment into a note in the main text
- Rename  loading to isLoading (#222)
- [SDK-2229] Migration guide (#223)
- Update dependencies
- Update the examples readme to include vercel deployment information
- Update README.md
- Restructure examples readme file
- Remove dangling pipe
- Update vercel URLs
- Add error management to useUser
- Re-add response conditional
- Move serverless-jwt/next to the example package
- Update dependencies
- run npm install after bringing in beta
- Revert Cypress
- Address review feedback
- Improve syntax highlighting in docs
- Remove unnecessary leading slash
- Disable x-powered-by header (#232)
- Prevent caching the `/me` request
- Use data-testid instead of ids
- Add selector for classic Lock
- Make smoke tests more robust
- Address review feedback
- Add login assert
- Bump @serverless-jwt/next to resolve some vulnerable deps (#235)
- AggregateError#message from `Issuer.discover` includes stack trace
- Release 1.0.0-beta.0
- Update dev dependencies
- Improve the CHANGELOG a little
- Add snippets to the logout/callback in migration guide
- Investigating bundle size
- fix test, missing types for tough-cookie
- Fix a couple of eslint errors from the eslint upgrade
- Update dev dependencies of kitchen sink example (#242)
- Avoid installing basic example dependencies for CI (#241)
- bump jose and openid-client
- Removing this advice for now to gather feedback
- Update configs to be able to run on vercel (#249)
- Added checkSession method to useUser hook [SDK-2256] (#246)
- withPageAuthrequired should not require env vars at build time
- Move single instance logic to another branch
- The CI environment has `process.envAUTH0_SECRET`
- Prevent custom session props being removed when refreshing AT (#257)
- Split the config into Next layer and Base layer specific configs
- Add useConfig hook (#262)
- Remove some unused options from express
- add nextConfig tests
- Remove attemptSilentLogin option
- Nextify the docs
- fix tests
- prettier removed some comments from the markdown
- Rename handler factories
- Remove callback route default
- Get the ID Token before deleting the session
- Update Vercel configuration docs (#266)
- Simplified configuration of custom login URL [SDK-2264] (#268)
- Add tslib to dependencies
- Update docs (#269)
- Release 1.0.0-beta.1 (#272)


[SDK-2059]: https://auth0team.atlassian.net/browse/SDK-2059
[SDK-2066]: https://auth0team.atlassian.net/browse/SDK-2066
